codec.go

```diff
@@ -52,27 +51,43 @@ func stringToBytes(s string) ([]byte, error) {
 	return b, nil
 }
 
-func bytesToString(b []byte) (ret string, err error) {
+func validateBytes(b []byte) (err error) {
```

Comment:
this function is not tested on its own. we should add it to all the tests wherever we can. it's critical that the behavior of this one is correct.